doc/source/whatsnew/v0.23.0.txt

```diff
@@ -711,6 +711,7 @@ Other API Changes
 - ``Categorical.fillna`` now validates its ``value`` and ``method`` keyword arguments. It now raises when both or none are specified, matching the behavior of :meth:`Series.fillna` (:issue:`19682`)
 - ``pd.to_datetime('today')`` now returns a datetime, consistent with ``pd.Timestamp('today')``; previously ``pd.to_datetime('today')`` returned a ``.normalized()`` datetime (:issue:`19935`)
 - :func:`Series.str.replace` now takes an optional `regex` keyword which, when set to ``False``, uses literal string replacement rather than regex replacement (:issue:`16808`)
+- ``DatetimeIndex.strftime`` and ``PeriodIndex.strftime`` now return an ``Index`` instead of a numpy array to be consistent with similar accessors (:issue:`20127`)
```

Comment:
can you add the :func: ref here

Reply:
in theory you should have to change some tests on ``Series.dt.strftime``, if you don't that means we are not fully testing this, can you check.

Reply:
But how would the output of ``Series.dt.strftime`` change when ``DatetimeIndex.strftime`` returns an ``Index`` instead of an array? It doesn't seem to ultimately affect anything in ``Series.dt.strftime`` from what I can tell.